frontend/catalyst/compilation_pipelines.py

```diff
@@ -837,6 +866,84 @@ def qjit(
             elements of this list are named sequences of MLIR passes to be executed. A ``None``
             value (the default) results in the execution of the default pipeline. This option is
             considered to be used by advanced users for low-level debugging purposes.
+        abstracted_axes (Sequence[Sequence[str]] or Dict[int, str] or Sequence[Dict[int, str]]):
+            An experimental option to specify dynamic tensor shapes.
+            This option affects the compilation of the annotated function.
+            Function arguments with ``abstracted_axes`` specified will be compiled to ranked tensors
+            with dynamic shapes.
```

Comment:
@erick-xanadu not blocking for PR, but I've noticed the `abstracted_axis` [documentation in the docstring](https://xanaduai-pennylane--366.com.readthedocs.build/projects/catalyst/en/366/code/api/catalyst.qjit.html) is now _really_ long!

What I would suggest is maybe doing something like this:

```rest
abstracted_axes (Sequence[Sequence[str]] or Dict[int, str] or Sequence[Dict[int, str]]):
    An experimental option to specify dynamic tensor shapes.
    This option affects the compilation of the annotated function.
    Function arguments with ``abstracted_axes`` specified will be compiled to ranked tensors
    with dynamic shapes. See below for more details and examples.

... (after the examples)

.. details::
    :title: Dynamic shape arrays and abstracted axes
    
    There are three ways to use ``abstracted_axes``; by passing a sequence of tuples, a
    dictionary, or a sequence of dictionaries. Passing a sequence of tuples:

    ...
```

Reply:
Not blocking for this PR though, and something I can change after